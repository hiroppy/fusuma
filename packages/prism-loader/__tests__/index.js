'use strict';

const { join } = require('path');
const loader = require('../src');

jest.mock('loader-utils');
const loaderUtils = require('loader-utils');

const prism = `
import Prism from 'prismjs';
Prism.highlightAll();
export { Prism };
`;
const dirPath = join(__dirname, './fixtures');

describe('prism-loader', () => {
  it('should add multiple languages', async () => {
    loaderUtils.getOptions.mockImplementation(() => {
      return {
        dirPath,
        plugins: [],
        theme: 'default',
      };
    });

    await loader.call(
      {
        async: () => (_, res) => {
          expect(res).toMatchSnapshot();
        },
      },
      prism
    );
  });

  it('should add multiple plugins', async () => {
    loaderUtils.getOptions.mockImplementation(() => {
      return {
        dirPath,
        plugins: ['line-highlight', 'line-numbers'],
        theme: 'default',
      };
    });

    await loader.call(
      {
        async: () => (_, res) => {
          expect(res).toMatchSnapshot();
        },
      },
      prism
    );
  });

  it('should add a theme', async () => {
    loaderUtils.getOptions.mockImplementation(() => {
      return {
        dirPath,
        plugins: [],
        theme: 'dark',
      };
    });

    await loader.call(
      {
        async: () => (_, res) => {
          expect(res).toMatchSnapshot();
        },
      },
      prism
    );
  });
});
