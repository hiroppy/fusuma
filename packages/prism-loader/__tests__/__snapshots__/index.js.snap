// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prism-loader should add a theme 1`] = `
"import Prism from \\"prismjs/components/prism-core.js\\";
import \\"prismjs/themes/prism-dark.css\\";
import \\"prismjs/components/prism-clike\\";
import \\"prismjs/components/prism-markup\\";
import \\"prismjs/components/prism-javascript\\";
import \\"prismjs/components/prism-css\\";
import \\"prismjs/components/prism-jsx\\";
Prism.highlightAll();
export { Prism };"
`;

exports[`prism-loader should add multiple languages 1`] = `
"import Prism from \\"prismjs/components/prism-core.js\\";
import \\"prismjs/themes/prism.css\\";
import \\"prismjs/components/prism-clike\\";
import \\"prismjs/components/prism-markup\\";
import \\"prismjs/components/prism-javascript\\";
import \\"prismjs/components/prism-css\\";
import \\"prismjs/components/prism-jsx\\";
Prism.highlightAll();
export { Prism };"
`;

exports[`prism-loader should add multiple plugins 1`] = `
"import Prism from \\"prismjs/components/prism-core.js\\";
import \\"prismjs/themes/prism.css\\";
import \\"prismjs/components/prism-clike\\";
import \\"prismjs/components/prism-markup\\";
import \\"prismjs/components/prism-javascript\\";
import \\"prismjs/components/prism-css\\";
import \\"prismjs/components/prism-jsx\\";
import \\"prismjs/plugins/line-highlight/prism-line-highlight.css\\";
import \\"prismjs/plugins/line-highlight/prism-line-highlight\\";
import \\"prismjs/plugins/line-numbers/prism-line-numbers.css\\";
import \\"prismjs/plugins/line-numbers/prism-line-numbers\\";
Prism.highlightAll();
export { Prism };"
`;
